File: unspeller/unspellerUnitTests/unspellerTests.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using NUnit.Framework;
using unspeller;

namespace unspellerUnitTests
{
    [TestFixture]
    public class unspellerTests
    {
        [Test]
        public void ChangeCaseFromLowerToUpperTest()
        {

            Assert.AreEqual('I', unspell.ChangeCase('i'));
        }

        [Test]
        public void ChangeCaseFromUpperToLowerTest()
        {
            Assert.AreEqual('i', unspell.ChangeCase('I'));
        }

        [Test]
        public void DontChangeCaseOfDifferentCultureTest() 
        {
            // DOS window can't handle all culture's letters
            // like Á specifically. It will become a regular A
            // and make the unspeller not realize the cultural
            // difference.
            Assert.AreEqual('Á', unspell.ChangeCase('Á'));
        }

        [Test]
        public void ChangeCharacterTest()
        {
            Assert.AreEqual("weke", unspell.ChangeCharacter("wake", 1, 'e'));
        }

        [Test]
        public void ChangeNoVowelWordTest()
        {
            var actual = unspell.ChangeVowels("blrg");

            Assert.AreEqual("blrg", actual);
        }

        [Test]
        public void ChangeAllVowelsTest()
        {
            // Not a practical example, but gets the job done
            var actual = unspell.ChangeVowels("eiouya", true);

            Assert.AreEqual("aeiouy", actual);
        }

        [Test]
        public void ChangeCapitalVowelTest()
        {
            // Not a practical example, but gets the job done
            var actual = unspell.ChangeVowels("E", true);

            Assert.AreEqual("A", actual);
        }

        [Test]
        public void ChangeWordTest()
        {
            var actual = unspell.ChangeVowels("sheep", true);

            Assert.AreEqual("shaep", actual);
        }

        [Test]
        public void GetRepeatingCharDataTest()
        {
            var actual = unspell.RepeatCharacters("job", true);

            Assert.AreEqual("joobbb", actual);
        }

        [Test]
        public void ChangeCapatalizeationOnWordTest()
        {
            var actual = unspell.ChangeCapatlizationOnWord("yUp", true);

            Assert.AreEqual("YuP", actual);
        }

        [Test]
        public void UnspellWordTest()
        {
            var actual = unspell.UnspellWord("yUp", true);

            Assert.AreEqual("AeePPP", actual);
        }
    }
}

File: RubySolution/test/unspeller_tests.rb
require "../lib/unspell"
require "test/unit"

#class LetterTree
#  attr_accessor :tree, :badWord
#end
 
class UnspellerTests < Test::Unit::TestCase
  def test_change_no_vowel_word
    actual = Unspell.change_vowels("blrg")
    assert_equal("blrg", actual)
  end
  
  def test_change_all_vowels
    actual = Unspell.change_vowels("eiouya", true)
    
    assert_equal("aeiouy", actual)
  end
  
  def test_change_capital_vowel
    actual = Unspell.change_vowels("E", true)
    
    assert_equal("A", actual)
  end
  
  def test_change_vowels_in_word
    actual = Unspell.change_vowels("sheep", true)
    
    assert_equal("shaep", actual)
  end
  
  def test_get_repeating_chars
    actual = Unspell.repeat_characters("job", true)
    
    assert_equal("joobbb", actual)
  end
  
  def test_change_capitalization_on_word
    actual = Unspell.change_capitalization("yUp", true)
    
    assert_equal("YuP", actual)
  end
  
  def test_unspell_word
    actual = Unspell.unspell_word("yUp", true)
    
    assert_equal("AeePPP", actual)
  end
end
